### PR TITLE
Fix broken links - July

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #24848. (Hide whitespace changes in diff.)

### CODE REVIEW

File: docs/breadcrumb/toc.yml [hunk 1]
```diff
@@ -464,7 +464,7 @@ items:
           topicHref: /dotnet/csharp/language-reference/proposals/csharp-8.0/index
         - name: C# 9.0 feature specifications
           tocHref: /dotnet/csharp/language-reference/proposals/csharp-9.0/
-          topicHref: /dotnet/csharp/language-reference/proposals/csharp-9.0/index
+          topicHref: /dotnet/csharp/language-reference/proposals/csharp-9.0/records
```
Comment (gewarren) [inline docs/breadcrumb/toc.yml:467]: @BillWagner I see that for C# 8, this is done through a redirection entry instead. LMK if you'd prefer that here.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

File: docs/breadcrumb/toc.yml (hunk 1 above)
Comment (BillWagner) [reply docs/breadcrumb/toc.yml:467]: @gewarren The reason I added the redirect for "index.md" was so hackable URLs work.  It's nice to have, but I don't have strong feelings about it.

**Comment by gewarren** [comment]
Thanks for reviewing David!

**Pull request merged**